src/taker/api2.rs

```diff
@@ -787,6 +788,16 @@ impl Taker {
         // Set swap params early so they're available for SwapDetails
         self.ongoing_swap_state.swap_params = swap_params.clone();
 
+        // Generate preimage for the swap
+        let mut preimage = [0u8; 32];
+        OsRng.fill_bytes(&mut preimage);
+
+        let unique_id = preimage[0..8].to_hex_string(bitcoin::hex::Case::Lower);
+        log::info!("Initiating coinswap with id : {}", unique_id);
+
+        self.ongoing_swap_state.active_preimage = preimage;
+        self.ongoing_swap_state.id = unique_id;
+
```

Comment:
This is not really part of the `choose_maker` routine. This can be moved into the parent swap function, just before calling the choose makers and processing rest of the swaps.

Reply:
moved to parent function